### DISCUSSION TITLE
Executable /usr/sbin/ipsec not found after building 6.0 from source

### DISCUSSION THREAD

**Discussion by robonuka** [strongswan/strongswan#2551] [General]
Hello guys!
I'm frustrated after succussfully built Strongswan 6.0 from source by these commands:

./configure --prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --with-ipsecdir=/usr/lib/strongswan <...some options...>
make
make install

Strongswan v5.9.14 was built successful. There is /usr/sbin/ipsec executable
Strongswan v6.0.0 was built successfully too, BUT there is NO /usr/sbin/ipsec and ipsec executables at all anywhere :(
How can I start Strongswan 6.0 as IPSecVPN server without /usr/sbin/ipsec ?

**Comment by strongX509** [comment]
The legacy stroke management interface has been deprecated for many years and has been replaced by the versatile [vici](https://docs.strongswan.org/docs/latest/plugins/vici.html) management interface. Thus with strongSwan 6.0, the stroke plugin is not enabled by default anymore and has to be built separately. See our announcement: [What’s New in strongSwan 6.0](https://docs.strongswan.org/docs/latest/news/whatsNew.html).

Thus you have to add the  **`--enable-stroke`** option to your **`./configure`** statement above.

**Reply by robonuka** [reply]
Thank you for advice! My goal is to build Strongswan 6.0 docker image for my own use.
For the 5.xx versions of Strongswan server I ran it in foreground as:
RUN  [ "/usr/sbin/ipsec start --nofork" ]
Which command I can run strongswan server in foreground in 6.0 version? I mean without "/usr/sbin/ipsec"

**Reply by strongX509** [reply]
In my [Dockerfile](https://github.com/strongX509/docker/blob/master/strongswan/Dockerfile) for building a strongSwan 6.0.0 docker image with the **`swanctl`**/**`vici`** control interface, I'm creating the symbolic link:
```
ln -s /usr/libexec/ipsec/charon charon && \
```
and with an interactive console window I manually start the IKEv2 daemon with the **`./charon &`** command in the background. Thus you could probably just use
```
RUN [ "/usr/libexec/ipsec/charon" ]
```
to start the daemon in the foreground.

**Reply by robonuka** [reply]
Thank you again!
I did even better) Look as 5.9.14 doing and did the same:
/usr/lib/strongswan/starter --daemon charon --nofork

**Reply by strongX509** [reply]
The **`starter`** command doesn't exist if the **`stroke`** interface is disabled.

**Reply by robonuka** [reply]
It's sad(, so I have to rebuild my next strongswan docker image to run without that 'starter'
